legend-depot-store-metrics/src/test/java/org/finos/legend/depot/store/metrics/TestMetricsServices.java

```diff
@@ -76,5 +76,14 @@ public void canGetMostRecentlyQueriedMetrics()
         Assert.assertEquals("2.2.0", metrics.get(1).getVersionId());
     }
 
-
+    @Test
+    public void canConsolidateQueryMetrics()
+    {
+        Assert.assertEquals(4, metricsStore.getAllStoredEntities().size());
+        metricsHandler.consolidateMetrics();
+        List<VersionQueryMetric> metrics = metricsStore.getAllStoredEntities();
+        Assert.assertEquals(2, metrics.size());
+        Assert.assertEquals("2.2.0", metrics.get(0).getVersionId());
+        Assert.assertEquals("1.0.0", metrics.get(1).getVersionId());
```

Comment:
we need test for metrics with same gav and different dates

Reply:
this is tested, if you look at the test data, though I would make it better pronounced